code_examples/sdk_examples/src/workshop/attester/generateKeypairs.ts

```diff
@@ -1,25 +1,50 @@
-import { mnemonicGenerate, mnemonicToMiniSecret } from '@polkadot/util-crypto'
-
 import * as Kilt from '@kiltprotocol/sdk-js'
+import {
+  blake2AsU8a,
+  keyExtractPath,
+  keyFromPath,
+  mnemonicGenerate,
+  mnemonicToMiniSecret,
+  naclBoxPairFromSecret,
+  sr25519PairFromSeed,
+} from '@polkadot/util-crypto';
+import { Keypair } from '@polkadot/util-crypto/types';
+import { generateAccount } from './generateAccount';
 
 export function generateKeypairs(mnemonic = mnemonicGenerate()) {
-  const authentication = Kilt.Utils.Crypto.makeKeypairFromSeed(
-    mnemonicToMiniSecret(mnemonic)
-  )
-  const encryption = Kilt.Utils.Crypto.makeEncryptionKeypairFromSeed(
-    mnemonicToMiniSecret(mnemonic)
-  )
-  const attestation = authentication.derive(
-    '//attestation'
-  ) as Kilt.KiltKeyringPair
-  const delegation = authentication.derive(
-    '//delegation'
-  ) as Kilt.KiltKeyringPair
+  const { account } = generateAccount(mnemonic);
+  // Authenticate presentations
+  const authentication = {
+    ...account.derive('//did//0'),
+    type: 'sr25519',
+  } as Kilt.KiltKeyringPair;
+
+  // Key used to attest transacations
+  const assertionMethod = {
+    ...account.derive('//did//assertion//0'),
+    type: 'sr25519',
+  } as Kilt.KiltKeyringPair;
+  // Key used for authority delgation
+  const capabilityDelegation = {
+    ...account.derive('//did//delegation//0'),
+    type: 'sr25519',
+  } as Kilt.KiltKeyringPair;
+
+  // Used to encrypt and decrypt messages
+  const keyAgreement: Kilt.NewDidEncryptionKey & Keypair = (function () {
```

Comment:
For the sake of clarity, I would like this closure to be reworked into a function above this one.